src/cr/cube/matrix/subtotals.py

```diff
@@ -370,6 +371,166 @@ def _subtotal_row(self, subtotal):
+        """
+        return cls(
+            base_values,
+            counts,
+            default_insertions,
+            dimensions,
+            diff_cols_nan,
+            diff_rows_nan,
+        )._subtotal_rows
+
+    @lazyproperty
+    def _subtotal_rows(self):
+        """(n_row_subtotals, n_cols) ndarray of subtotal rows."""
+        subtotals = self._row_subtotals
+
+        if len(subtotals) == 0:
+            return np.empty((0, self._ncols))
+
+        return np.vstack(
+            [
+                self._subtotal_row(subtotal, default)
+                for subtotal, default in zip(subtotals, self._default_insertions)
+            ]
+        )
+
+    @classmethod
+    def subtotal_columns(
+        cls,
+        base_values,
+        counts,
+        default_insertions,
+        dimensions,
+        diff_cols_nan=False,
+        diff_rows_nan=False,
+    ):
+        """Return (n_column_subtotals, n_base_rows) ndarray of subtotal columns."""
+        return cls(
+            base_values,
+            counts,
+            default_insertions,
+            dimensions,
+            diff_cols_nan,
+            diff_rows_nan,
+        )._subtotal_columns
+
+    @lazyproperty
+    def _subtotal_columns(self):
+        """(n_rows, n_col_subtotals) matrix of subtotal columns."""
+        subtotals = self._column_subtotals
+        if len(subtotals) == 0:
+            return np.empty((self._nrows, 0))
+        return np.hstack(
+            [
+                self._subtotal_column(subtotal, default).reshape(self._nrows, 1)
+                for subtotal, default in zip(subtotals, self._default_insertions.T)
+            ]
+        )
+
+    def _subtotal_column(self, subtotal, default):
+        """Return (n_rows,) ndarray of values for `subtotal` column."""
+        if (
+            self._dimensions[-1].dimension_type == DT.CAT_DATE
+            and len(subtotal.subtrahend_idxs) > 0
+            and len(subtotal.addend_idxs) > 0
+        ):
+            if self._multiple_subtrahends_or_addends(subtotal):
+                return self._nan_subtotals(axis=0)
+            base_addend_sum = np.sum(self._base_values[:, subtotal.addend_idxs], axis=1)
+            base_subtrahend_sum = np.sum(
+                self._base_values[:, subtotal.subtrahend_idxs], axis=1
+            )
+            counts_addend_sum = np.sum(self._counts[:, subtotal.addend_idxs], axis=1)
+            counts_subtrahend_sum = np.sum(
+                self._counts[:, subtotal.subtrahend_idxs], axis=1
+            )
+            return (counts_addend_sum / base_addend_sum) - (
+                counts_subtrahend_sum / base_subtrahend_sum
+            )
+
+        return default
+
+    def _subtotal_row(self, subtotal, default):
+        """Return (n_cols,) ndarray of values for `subtotal` row."""
+
+        if (
+            self._dimensions[0].dimension_type == DT.CAT_DATE
+            and len(subtotal.subtrahend_idxs) > 0
+            and len(subtotal.addend_idxs) > 0
```

Comment:
Maybe this should just be cat date with any subtrahends? Seems weird that having 2 subtrahends and 0 addend would work the old way, but 2 subtrahends and 1 addend would be NaN?